tool/src/cpp2/header.rs

```diff
@@ -80,11 +70,39 @@ impl Header {
             path,
             includes: BTreeSet::new(),
             decl_include: None,
-            forwards: BTreeSet::new(),
+            forwards: BTreeMap::new(),
             body: String::new(),
             indent_str: "  ",
         }
     }
+
+    fn forward_for(def: TypeDef, ty_name_unnamespaced: &str) -> Forward {
+        match def {
```

Comment:
Thought: I kind-of like the old style of putting the `Forward` type directly in `gen_type_name` to avoid the double match

Reply:
Yeah I wasn't sure of this but decided to go this way so that the rest of the code doesn't need to worry about the header's internal ontology of forward kinds.